### ISSUE TITLE
Continuous changing agent colour and radius during visualisation

### ISSUE THREAD

**Issue by guyguyguyguyguyguyguy** [projectmesa/mesa#731]
**What's the problem this feature will solve?**
I would like to see agents changing over time in response to changes in their states.

**Describe the solution you'd like**
I am looking to alter the look (colour/radius) of agents after each step in the simulation, dependant on their states changing

**Additional context**
Thanks :)

**Comment by guyguyguyguyguyguyguy** [comment]
I think a possible solution is to remove the old agents and create new ones at each step with the updated states. These will then be represented differently in the visualisation if they have met certain conditions. However, I was wondering if there was a more convenient way

**Comment by tpike3** [comment]
I would not recommend removing the agent and creating a new one. I would focus on the visualization code. From the read the docs [visualization section](https://mesa.readthedocs.io/en/master/tutorials/adv_tutorial.html), the visualization is a loop so I would add conditionals to get the effect you want you want,

For example: 

    def agent_portrayal(agent):
         if state == "x": ` 
              portrayal = {"Shape": "circle",
              "Color": "red",
              "Filled": "true",
              "Layer": 0,
              "r": 0.5}
          elif state == "y":
               portrayal = {"Shape": "square",
               "Color": "blue",
               "Filled": "true",
               "Layer": 0,
               "r": 0.7} 
        return portrayal


Hope this helps, 
Tom

**Comment by guyguyguyguyguyguyguy** [comment]
Hi Tom,
Thank you very much for your response. I have tried using a conditional portrayal, however, the conditional only seems to act during initialisation of the visualisation, and, does not alter the agents representation at each step the model is run.  

Lets say I have an agent state, agent.state, at initialisation of the visualisation agent.state = 0, which gives this agent a radius of 0.1, "r":0.1. Then at the next step of the model said agent receives money, agent.state= 0.5. What I would like is for this agent's radius to increase to 0.5 ("r":0.5) during the visualisation. 

I tried to do this using a conditional portrayal but for me it did not work. I tried something like this:

```
def agent_portrayal(agent):

    portrayal = {"Shape": "circle",
                "Color": "red",
                "Filled": "true",
                "Layer": 1}
    
    if agent.state < 0.5:
        portrayal["r"] = 0.1
    elif agent.state < 1:
        portrayal["r"] = 0.5
    else:
        portrayal["r"] = 1
    
    return portrayal

```
